gps/prune.go

```diff
@@ -19,6 +20,13 @@ type PruneOptions uint8
 // PruneProjectOptions is map of prune options per project name.
 type PruneProjectOptions map[ProjectRoot]PruneOptions
 
+// RootPruneOptions represents the root prune options for the project.
+// It contains the global options and a map of options per project.
+type RootPruneOptions struct {
```

Comment:
i'm looking at this and wondering if we can leave `PruneOptions` in gps, but move the `RootPruneOptions` concept up and out to the root `dep` package. we might optionally then also remove the `PruneProjectOptions` type and instead just take a `map[ProjectRoot]PruneOptions` anywhere that we currently take a `PruneProjectOptions`.

i realize this is reaching back into the last PR, but now that i'm really thinking about the added surface area, it seems like these additional abstractions aren't strictly necessary in gps, and i'm not sure what we get out of having them here, vs. keeping them in the same package with all the root-vs-specific-project prune options logic that's in `dep/manifest.go`.

i'm really on the fence about this, though, and could probably be convinced to keep them here if you think it's important.

Reply:
It really boils down to what we want `gps` to offer. If `gps` will handle vendor verification then this should stay here (the structs can be refactored though).

The main advantage of `RootPruneOptions` is that we pass one less parameter to `WriteDepTree` (we could pass a `PruneOptions` and a `map[ProjectRoot]PruneOptions`). This was the original case, but I thought having a `RootPruneOptions` is cleaner.

We can precompute `map[ProjectRoot]PruneOptions` for all projects in the lock before calling `WriteDepTree` and pass it the map instead. I'm impartial to both tbh.